src/app/movies/movies.component.ts

```diff
@@ -21,6 +22,8 @@ export class MoviesComponent {
   }
 
   parseUserInput(movieInput: string): Movie {
+    const getVideoMetadata = getVideoId.getVideoId(movieInput);
+    console.log(getVideoMetadata);
     return {
       type: 'youtube | vimeo',
```

Comment:
to już jak masz te dane to nie zwracaj tutaj zahardkodowanego typu i id tylko zwróć to, co uzyskałeś po rozpoznaniu

Reply:
Czyli po prostu w return zostawię tylko link: movieInput który mi zwraca link. Z libki jest zwracany id i service.  Chyba że jakoś kombinować żeby w type zamiast 'youtube | vimeo' pokazywała się wartość service a w id pokazywała się wartość id pobierana przez libke ?

Reply:
Z tym że jeśli chcę zmienić wartości zwracane to muszę ingerować w plik interfaces.ts i tam zostawić tylko wartość link.

Reply:
tka własnie masz "zakombinować" - chcesz żeby funkcja parseInput zwróciła Ci obiekt opisujący ten film, no nie? Id: '' to żadna informacja, każdy film miałby to samo. Więc zrób tak, żeby to id to było prawdziwe id filmu, a typ żeby był prawdziwym serwisem pochodzenia. W związku z tym interfejs jest ok, nic tam nie zmieniaj

Reply:
Interface opisuje Ci jakie properties będą na obiekcie i jakiego będą typu. O ile pamiętam masz tam, że id to string (dowolny) a type to 'youtube' | 'vimeo' (czyli albo jedno albo drugie, już nie dowolny string). Wartości zwracane przez funkcję będą do tego interfejsu pasowały, a przynajmniej powinny (jesli by np funkcja zwracała Vimeo zamiast vimeo, to trzeba by zrobić toLowerCase tutaj albo zmienić funkcję).

Reply:
Spojrzałem na ten interfejs i masz błąd który przeoczyłem a który nie wyrzuca nic w kompilatorze. Masz w interface Movie napisane type: 'youtube | vimeo'; a to oznacza, że property type obiektu typu Movie musi szawsze być stringiem 'youtube | vimeo'. Zmień interface na type: 'youtube' | 'vimeo', tak żeby property type było tym lub tym ;)